### PR TITLE
Allow rustls for TLS instead of OpenSSL

### PR DESCRIPTION
**Pull request by edbo**
* Update Reqwest version
* Update Cargo.toml to allow using Reqwest rustls instead of OpenSSL
* Update the README to explain how to use rustls

### CODE REVIEW

**Comment by edbo** [comment]
If everything is OK please merge this one first because it fixes the tests and runs them in the CI: https://github.com/PropelAuth/rust/pull/37

Then this one can be merged and it's tests pass here:

https://github.com/ShiftControl-io/rust/actions/runs/8581346252/job/23518285809

**Comment by mrmauer** [comment]
Hey again, @edbo! Could you go into more detail as to why you want to use rustls over the default?

File: Cargo.toml [hunk 1]
```diff
@@ -25,16 +25,19 @@ uuid = { version = "^1.0", features = ["serde"] }
...
 [dev-dependencies]
 openssl = "0.10.43"
 
 [features]
+default = ["reqwest/default", "__reqwest"]
+rustls = ["__reqwest", "reqwest/rustls-tls", "reqwest/http2", "reqwest/charset"]
 axum06 = ["dep:axum_06", "dep:tower"]
 axum07 = ["dep:axum_07", "dep:tower"]
 actix4 = ["dep:actix-web"]
+__reqwest = ["reqwest/json"]
```
Comment (mrmauer) [inline Cargo.toml:40]: Is there a reason you're removing the `multipart` feature of `reqwest` from the default?

**Comment by edbo** [comment]
@mrmauer It builds without it so I tried to find where multipart was used and couldn't find it. I guess it also surprised me that your API would have multipart APIs but I could be wrong on that, obviously you guys would know better.

For reference, it was introduced with this change: https://github.com/PropelAuth/rust/commit/895358e7d943640162a3bb25d76bf714c3d53f92

Anyway, let me know if you'd prefer I add multipart back as there could be good reason for it.

**Comment by mrmauer** [comment]
Our apis support multi part for image uploads (eg profile pictures). Let's add that back. 

I'm broadly open to the pass through feature flag to enable rustls, but I want to circle back and do some gut checks next week.

File: Cargo.toml (hunk 1 above)
Comment (edbo) [reply Cargo.toml]: @mrmauer I've added multipart back now.

**Comment by edbo** [comment]
@mrmauer I've just rebased this. Any chance of getting it merged this week?

**Review by mrmauer — COMMENTED** [review]
Hey @edbo. Tested this and it LGTM. A little note on the copy in the README to make it up-to-date and web-framework-agnostic, and then I can merge.

I'd say we can probably release this by EOW, but if not then early next week.

**Review by mrmauer — APPROVED** [review]
LGTM

**Pull request merged**